src/System.Net.Security/src/System/Net/_SSPIWrapper.cs

```diff
@@ -0,0 +1,873 @@
+            {
+                lock (SecModule)
+                {
+                    if (SecModule.SecurityPackages == null)
+                    {
+                        int moduleCount = 0;
+                        SafeFreeContextBuffer arrayBaseHandle = null;
+                        try
+                        {
+                            int errorCode = SecModule.EnumerateSecurityPackages(out moduleCount, out arrayBaseHandle);
+                            GlobalLog.Print("SSPIWrapper::arrayBase: " + (arrayBaseHandle.DangerousGetHandle().ToString("x")));
+                            if (errorCode != 0)
+                            {
+                                throw new Win32Exception(errorCode);
+                            }
+                            SecurityPackageInfoClass[] securityPackages = new SecurityPackageInfoClass[moduleCount];
+                            if (Logging.On) Logging.PrintInfo(Logging.Web, SR.net_log_sspi_enumerating_security_packages);
+                            int i;
+                            for (i = 0; i < moduleCount; i++)
+                            {
+                                securityPackages[i] = new SecurityPackageInfoClass(arrayBaseHandle, i);
+                                if (Logging.On) Logging.PrintInfo(Logging.Web, "    " + securityPackages[i].Name);
+                            }
+                            SecModule.SecurityPackages = securityPackages;
+                        }
+                        finally
+                        {
+                            if (arrayBaseHandle != null)
+                            {
+                                arrayBaseHandle.Dispose();
+                            }
+                        }
+                    }
+                }
+            }
+            GlobalLog.Leave("EnumerateSecurityPackages");
+            return SecModule.SecurityPackages;
+        }
+
+        internal static SecurityPackageInfoClass GetVerifyPackageInfo(SSPIInterface secModule, string packageName)
+        {
+            return GetVerifyPackageInfo(secModule, packageName, false);
+        }
+
+        internal static SecurityPackageInfoClass GetVerifyPackageInfo(SSPIInterface secModule, string packageName, bool throwIfMissing)
+        {
+            SecurityPackageInfoClass[] supportedSecurityPackages = EnumerateSecurityPackages(secModule);
+            if (supportedSecurityPackages != null)
+            {
+                for (int i = 0; i < supportedSecurityPackages.Length; i++)
+                {
+                    if (string.Compare(supportedSecurityPackages[i].Name, packageName, StringComparison.OrdinalIgnoreCase) == 0)
+                    {
+                        return supportedSecurityPackages[i];
+                    }
+                }
+            }
+
+            if (Logging.On) Logging.PrintInfo(Logging.Web, SR.Format(SR.net_log_sspi_security_package_not_found, packageName));
```

Comment:
Actually I don't agree with that style: it is error prone and can create the impression that indented lines are actually part of the if block.

It's not in rules within https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md .

I would rather have:
1. If the `if` statement can appear on a single line it can miss braces.
2. Any `if` statement that doesn't fit on a line should _always_ have braces.